### PR TITLE
[Intel MKL] Making BFloat16 MatMul & BatchMatMul in DNNL 1.2 conditio…

### PR DESCRIPTION
**Pull request by nhasabni**
…nally compilable

This PR makes BFloat16 support for MatMul and BatchMatMul conditionally compilable by
removing macros that were guarding DNNLv1.2 specific code. It also makes necessary
changes to Gemm helper functions in MklBatchMatMulOp to handle type-specific helper
functions appropriately.

### CODE REVIEW

**Comment by nhasabni** [comment]
Hi @penpornk, I've added an explanation in the PR description. Let me know if that is not enough. I can add more details. Thanks.

File: tensorflow/core/kernels/mkl_batch_matmul_op.cc [hunk 1]
```diff
@@ -252,27 +254,27 @@ class BatchMatMulMkl : public OpKernel {
...
-        reinterpret_cast<const bfloat16**>(B_Array), ldb_Array, beta_Array,
-        reinterpret_cast<bfloat16**>(C_Array), ldc_Array, group_count,
-        group_size);
+    dnnl_gemm_batch<bfloat16>(TransA_Array, TransB_Array, M_Array, N_Array,
+                              K_Array, alpha_Array, A_Array, lda_Array, B_Array,
+                              ldb_Array, beta_Array, C_Array, ldc_Array,
+                              group_count, group_size);
   }
-#endif  // ENABLE_MKLDNN_V1_2
+#endif  // ENABLE_INTEL_MKL_BFLOAT16 && ENABLE_INTEL_MKL_BFLOAT16
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_batch_matmul_op.cc]: Nit:
```suggestion
#endif  // ENABLE_MKLDNN_V1 && ENABLE_INTEL_MKL_BFLOAT16
```

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR! Minor nit please.

File: tensorflow/core/kernels/mkl_batch_matmul_op.cc (hunk 1 above)
Comment (nhasabni) [reply tensorflow/core/kernels/mkl_batch_matmul_op.cc]: Oops. Not sure how I missed it. :)

**Review by penpornk — APPROVED** [review]
Thank you for the quick fix!

**Review by rthadur — APPROVED** [review]

**Pull request merged**